core/tests/test_utils.py

```diff
@@ -80,6 +81,51 @@ def empty_environ():
         os.environ['HTTP_HOST'], os.environ['SERVER_PORT'])
 
 
+def get_filepath_from_filename(filename, rootdir):
+    """Returns filepath using the filename.
+
+    Args:
+        filename: str. The name of the file.
+        rootdir: str. The directory to search the file in.
+
+    Returns:
+        str. The path of the file if file is found otherwise
+            an empty string.
+    """
+    # This is required since error files are served according to error status
+    # code. The file served is error-page.mainpage.html but it is compiled
+    # and stored as error-page-{status_code}.mainpage.html.
+    # So, we need to swap the name here to obtain the correct filepath.
+    if filename.startswith('error-page'):
+        filename = 'error-page.mainpage.html'
+    for root, _, files in os.walk(rootdir):
+        for name in files:
+            if name == filename:
+                return os.path.join(root, filename)
+    return ''
```

Comment:
None is more semantically correct. (Update docstring typeinfo accordingly.)

Reply:
Updated.